core/incremental/operator.rs

```diff
@@ -581,34 +582,189 @@ impl IncrementalOperator for FilterOperator {
+        // Set to read-only mode and disable auto-commit since we're only evaluating expressions
+        internal_conn.query_only.set(true);
+        internal_conn.auto_commit.set(false);
+
+        let temp_syms = SymbolTable::new();
+
+        // Extract columns from SELECT statement
+        let columns = if let OneSelect::Select(ref select_stmt) = &*select.body.select {
+            let mut columns = Vec::new();
+            for result_col in &select_stmt.columns {
+                match result_col {
+                    ResultColumn::Expr(expr, alias) => {
+                        let alias_str = if let Some(As::As(alias_name)) = alias {
+                            Some(alias_name.as_str().to_string())
+                        } else {
+                            None
+                        };
+                        // Try to compile the expression (handles both columns and complex expressions)
+                        match CompiledExpression::compile(
+                            expr,
+                            &input_column_names,
+                            schema,
+                            &temp_syms,
+                            internal_conn.clone(),
+                        ) {
+                            Ok(compiled) => {
+                                columns.push(ProjectColumn {
+                                    expr: expr.clone(),
+                                    alias: alias_str,
+                                    compiled,
+                                });
+                            }
+                            Err(_) => {
+                                // If compilation fails, skip this column for now
+                                // In the future we might want to handle this better
+                            }
+                        }
+                    }
+                    ResultColumn::Star => {
+                        // Select all columns - create trivial column references
+                        for name in &input_column_names {
+                            // Create an Id expression for the column
+                            let expr = Expr::Id(Name::Ident(name.clone()));
+                            // This should always compile successfully as a trivial column reference
+                            if let Ok(compiled) = CompiledExpression::compile(
+                                &expr,
+                                &input_column_names,
+                                schema,
+                                &temp_syms,
+                                internal_conn.clone(),
+                            ) {
+                                columns.push(ProjectColumn {
+                                    expr,
+                                    alias: None,
+                                    compiled,
+                                });
+                            }
+                        }
+                    }
+                    _ => {
```

Comment:
I think it would be great to improve on our previous bad decisions in tursodb in general and just return parse errors on any unsupported features. There's two big advantages to this:

1. It removes confusion from anyone trying to use the feature, and obviously makes returning incorrect results far less likely, and
2. It makes it easier for developers to see at a glance which features we still need to implement